planner/core/rule_aggregation_elimination.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"math"
+
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/expression/aggregation"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+)
+
+type aggregationEliminator struct {
+	aggregationEliminateChecker
+}
+
+type aggregationEliminateChecker struct {
```

Comment:
remove `aggregationEliminateChecker` and `s/aggregationEliminateChecker/aggregationEliminator/`

Reply:
But `aggregationEliminateChecker` also used in `aggregation push down`